DeviceDetector.php

```diff
@@ -191,105 +191,114 @@ public function __construct($userAgent = '')
         $this->addBotParser(new Bot());
     }
 
-    public function __call($methodName, $arguments)
+    /**
+     * @param string $methodName
+     * @param array  $arguments
+     *
+     * @return bool
+     */
+    public function __call(string $methodName, array $arguments): bool
     {
-        foreach (DeviceParserAbstract::getAvailableDeviceTypes() as $deviceName => $deviceType) {
-            if (strtolower($methodName) == 'is' . strtolower(str_replace(' ', '', $deviceName))) {
-                return $this->getDevice() == $deviceType;
+        foreach (AbstractDeviceParser::getAvailableDeviceTypes() as $deviceName => $deviceType) {
+            if (\strtolower($methodName) === 'is' . \strtolower(\str_replace(' ', '', $deviceName))) {
+                return $this->getDevice() === $deviceType;
             }
         }
 
         foreach (self::$clientTypes as $client) {
-            if (strtolower($methodName) == 'is' . strtolower(str_replace(' ', '', $client))) {
-                return $this->getClient('type') == $client;
+            if (\strtolower($methodName) === 'is' . \strtolower(\str_replace(' ', '', $client))) {
+                return $this->getClient('type') === $client;
             }
         }
 
-        throw new \BadMethodCallException("Method $methodName not found");
+        throw new \BadMethodCallException("Method {$methodName} not found");
     }
 
     /**
      * Sets the useragent to be parsed
      *
      * @param string $userAgent
      */
-    public function setUserAgent($userAgent)
+    public function setUserAgent(string $userAgent): void
     {
-        if ($this->userAgent != $userAgent) {
+        if ($this->userAgent !== $userAgent) {
             $this->reset();
         }
-        $this->userAgent = $userAgent;
-    }
 
-    protected function reset()
-    {
-        $this->bot = null;
-        $this->client = null;
-        $this->device = null;
-        $this->os = null;
-        $this->brand = '';
-        $this->model = '';
-        $this->parsed = false;
+        $this->userAgent = $userAgent;
     }
 
     /**
-     * @param ClientParserAbstract|string $parser
+     * @param AbstractClientParser|string $parser
+     *
      * @throws \Exception
      */
-    public function addClientParser($parser)
+    public function addClientParser($parser): void
     {
-        if (is_string($parser) && class_exists('DeviceDetector\\Parser\\Client\\' . $parser)) {
+        if (\is_string($parser) && \class_exists('DeviceDetector\\Parser\\Client\\' . $parser)) {
             $className = 'DeviceDetector\\Parser\\Client\\' . $parser;
-            $parser = new $className();
+            $parser    = new $className();
```

Comment:
currently assignments are aligned. The used coding standard defines that and it can easily be adjusted automatically using `vendor/bin/phpcbf --standard=phpcs.xml .`